src/netCDF4/_netCDF4.pyx

```diff
@@ -1353,6 +1354,42 @@ to HDF5 version 1.8.x is highly recommended
 (see https://github.com/Unidata/netcdf-c/issues/250)."""
     warnings.warn(msg)
 
+if __has_set_alignment__:
+    def get_alignment():
```

Comment:
Probably should wrap in IF HAS_SET_ALIGNMENT and either return None if not supported

Reply:
Something like the following?

```python
diff --git a/src/netCDF4/_netCDF4.pyx b/src/netCDF4/_netCDF4.pyx
index 818188f9..91e843df 100644
--- a/src/netCDF4/_netCDF4.pyx
+++ b/src/netCDF4/_netCDF4.pyx
@@ -1354,7 +1354,7 @@ to HDF5 version 1.8.x is highly recommended
 (see https://github.com/Unidata/netcdf-c/issues/250)."""
     warnings.warn(msg)
 
-if __has_set_alignment__:
+if HAS_SET_ALIGNMENT:
     def get_alignment():
         """
     **`get_alignment()`**
@@ -1388,6 +1388,17 @@ if __has_set_alignment__:
 
         ierr = nc_set_alignment(thresholdp, alignmentp)
         _ensure_nc_success(ierr)
+else:
+    def get_alignemnt():
+        raise RuntimeError(
+            "This function requires netcdf4 4.9.0+ to be used at compile time"
+        )
+
+    def set_alignemnt(threshold, alignment):
+        raise RuntimeError(
+            "This function requires netcdf4 4.9.0+ to be used at compile time"
+        )
+
 
 
 # numpy data type <--> netCDF 4 data type mapping.
```

Reply:
yes, but if and else have to be capitalized (https://docs.cython.org/en/latest/src/userguide/language_basics.html#conditional-compilation)